src/Microsoft.Diagnostics.DebugServices.Implementation/SymbolService.cs

```diff
@@ -157,7 +299,39 @@ public void DisableSymbolStore()
         /// <returns>path to the downloaded file either in the cache or in the temp directory or null if error</returns>
         public string DownloadFile(SymbolStoreKey key)
         {
-            return SymbolReader.GetSymbolFile(key);
+            string downloadFilePath = null;
+
+            if (IsSymbolStoreEnabled)
+            {
+                using SymbolStoreFile file = GetSymbolStoreFile(key);
```

Comment:
In the chain of SymbolStore's the cache should be first and it does return the file from the cache before it attempts to use the  HttpSymbolStore in the chain to download. 

The threading model has to be synchronous (I assume this everywhere) to run under windbg/dbgeng which has a giant lock around the engine. It would cause deadlocks. Adding async to even this API would spread async everywhere (like a disease) and the native SOS code calling this would still have to do the wait.

The sync over async here could cause problems when upper layers are async (dotnet-dump starts async at the very upper layer but it doesn't do anything else asynchronously when it executes a command that could "reenter" the framework).  If we ever implement this framework on VS (someday), this could cause a problem.  I thought about changing symstore to provide sync interfaces for the "GetFile" function but that would require a lot of changes.

Reply:
synchronizing should be a limitation of the host, this layer should probably know it's potentially a network call so it needs to not block.